lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1332,16 +1332,19 @@ void HdVP2Mesh::_UpdateDrawItem(
         }
     }
 
+    auto drawItemVisible = [&]() {
+        return drawItem->GetVisible() && drawScene.DrawRenderTag(renderIndex.GetRenderTag(GetId()));
+    };
+
     if (itemDirtyBits & (HdChangeTracker::DirtyVisibility | HdChangeTracker::DirtyRenderTag)) {
-        drawItemData._enabled
-            = drawItem->GetVisible() && drawScene.DrawRenderTag(renderIndex.GetRenderTag(GetId()));
+        drawItemData._enabled = drawItemVisible();
         stateToCommit._enabled = &drawItemData._enabled;
     }
 
     if (isDedicatedSelectionHighlightItem) {
         if (itemDirtyBits & DirtySelectionHighlight) {
             const bool enable =
-                (_selectionState != kUnselected) && drawItem->GetVisible();
+                (_selectionState != kUnselected) && drawItemVisible();
```

Comment:
Interestingly my change for the UsdSkel geometry corruption issue is also fixing this part.

Reply:
So my fix will also include a check on the points array. If it is empty, the render item will be disabled. I will push my branch soon so that we can coordinate on the conflict.